src/components/posts/Form/Form.js

```diff
@@ -15,12 +22,25 @@ export default class Form extends React.Component {
     this.setState({ [name]: value })
   }
 
-  handleSubmit (e) {
+  async handleSubmit (e) {
     e.preventDefault()
-    this.props.onSubmit(this.state)
+    let response = await this.props.onSubmit(this.props.user, {content: this.state.content, emotion: this.state.emotion})
```

Comment:
this.props.user is undefined for the edit route. Check how you are passing params for that route.